### ISSUE TITLE
Preview Fails when footnote added.

### ISSUE THREAD

**Issue by kochie** [careteditor/caret#360]
The following error has occurred:
```
{
    "message": "Cannot read property 'type' of undefined",
    "version": "2.1.0",
    "os": "Linux 4.10",
    "path": "node_modules/markdown-it-footnote/index.js",
    "code": "1].type === 'paragraph_close') ",
    "line": 335,
    "column": 48,
    "stack": [
        "at Array.footnote_tail (node_modules/markdown-it-footnote/index.js:335:48)",
        "at Core.process (node_modules/markdown-it/lib/parser_core.js:51:13)",
        "at MarkdownIt.parse (node_modules/markdown-it/lib/index.js:519:13)",
        "at /opt/caret/resources/app.asar/packages/Preview/markdown-it-page-break.js:5:709",
        "at Array.forEach ((native)",
        "at toTokens (packages/Preview/markdown-it-page-break.js:5:586)",
        "at Array.<anonymous> (packages/Preview/markdown-it-page-break.js:5:1095)",
        "at Core.process (node_modules/markdown-it/lib/parser_core.js:51:13)",
        "at MarkdownIt.parse (node_modules/markdown-it/lib/index.js:519:13)",
        "at MarkdownIt.render (node_modules/markdown-it/lib/index.js:539:36)"
    ]
}
```

**Comment by astoilkov** [comment]
Hi @kochie,

I tried to replicate the problem by adding a footnote to the document and opening the preview but wasn't able to reproduce it. Here is what I tried:

```
Some long sentence. [^footnote]

[^footnote]: Test, [Link](https://google.com).
```

Do you have the text that causes this error?

Thanks in advance.

**Comment by manuelbua** [comment]
I tried to replicate it as well, but it's working fine for me on `2.1.1`/ArchLinux64.

**Comment by kochie** [comment]
Sorry, should of given more information (Had an assignment due). I'll attach the relevent [file](https://drive.google.com/file/d/0B4vcQMnGeBujM1pvSFhKS0ZZVVE/view?usp=sharing).
I believe the issue will be relating to the amount of math text I have on the page. Or possibly the table of contents.

**Comment by astoilkov** [comment]
Thanks for the document. We successfully replicated the problem and fixed it internally. The next release will include this change.

**Comment by astoilkov** [comment]
This is now fixed in the latest 2.1.3 release.

Closing this issue. Thank you for reporting and for the additional information.
